tests/unit/test_finder.py

```diff
@@ -34,6 +34,15 @@ def test_no_partial_name_match(data):
     assert found.url.endswith("gmpy-1.15.tar.gz"), found
 
 
+@patch('pip.index.os.path.exists')
+def test_tilde(data):
```

Comment:
Both good points and I think I had a smell about this but I never acted on it. Thanks for the kick in the pants. Updated.

```python
@patch(
    'pip.index.os.path.exists',
    return_value=True  # b/c we only use tilde expanded version if it exists
)
def test_tilde(data):
    """Finder can accept a path with ~ in it and will normalize it."""
    finder = PackageFinder(['~/python-pkgs'], [], session=PipSession())
    req = InstallRequirement.from_line("gmpy")
    with pytest.raises(DistributionNotFound):
        finder.find_requirement(req, False)
```